Neuroscan Support

yjmantilla opened the pull request:
PR Description
--------------

PR to add support of .cnt files.

Closes #922 



Output of make pep just said:

```
"Running flake8"
0
"Running pydocstyle"
"Checking MANIFEST.in"
lists of files in version control and sdist match
```

I ran the tests locally and it said ``237 passed``  (at the end).

Documentation build ended with:

```
Sphinx-Gallery successfully executed 13 out of 13 files subselected by:

    gallery_conf["filename_pattern"] = '^((?!sgskip).)*$'
    gallery_conf["ignore_pattern"]   = '__init__\\.py'

after excluding 0 files that had previously been run (based on MD5).

embedding documentation hyperlinks...
embedding documentation hyperlinks for auto_examples... [100%] update_bids_datasets.htmlhtml
make[1]: Leaving directory 'Y:/code/mne-bids/doc'
cd doc/ && make view
make[1]: Entering directory 'Y:/code/mne-bids/doc'
make[1]: Leaving directory 'Y:/code/mne-bids/doc'
```


Let me know if anything else is required, like an specific test for cnt files.

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [ ] All comments resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [ ] PR title starts with [MRG]
- [ ] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/main/doc/whats_new.rst) is updated
- [ ] PR description includes phrase "closes <#issue-number>"

agramfort reviewed (commented):
looks good

@yjmantilla can you add your name in this file

https://github.com/mne-tools/mne-bids/blob/main/doc/whats_new.rst

with a description of the change?

sappelhoff reviewed (commented):
Aaaand could you add a small test please?

yjmantilla commented:
@agramfort Sure, im a bit busy but as soon as I can I will do that.

@sappelhoff  Yeah I could. Here is a public cnt dataset which may suffice:

https://figshare.com/articles/dataset/The_original_EEG_data_for_driver_fatigue_detection/5202739

Should I add a single or a couple of this cnts file to mne.datasets.testing?

Should I be following an example test? Im guessing it somehow has to do with

https://github.com/mne-tools/mne-bids/blob/4c0755c6eb567c7840d70acf06f22201c5e75c65/mne_bids/tests/test_write.py#L118-L123

Would it suffice adding the cnt to the testing datasets and append the CNT case to this list?

agramfort commented:
maybe look at test functions like test_set or or test_bdf in test_write.py?

>

sappelhoff commented:
> Would it suffice adding the cnt to the testing datasets and append the CNT case to this list?

In the [`MNE-testing-data` there is a CNT file called `scan41_short.cnt`](https://github.com/mne-tools/mne-testing-data/tree/master/CNT). That one will probably suffice for a short/tiny test.

> Sure, im a bit busy but as soon as I can I will do that.

Cool, please add info about this PR under "enhancements" then. Note, that you also need to add your name to https://github.com/mne-tools/mne-bids/blob/main/doc/authors.rst in order for the link to your name to work.

> maybe look at test functions like test_set or or test_bdf in test_write.py?

agreed, or maybe this one:

https://github.com/mne-tools/mne-bids/blob/4c0755c6eb567c7840d70acf06f22201c5e75c65/mne_bids/tests/test_write.py#L1150

yjmantilla commented:
Hi @sappelhoff  and @agramfort , sorry for the delay. I added the tests and updated what's new.

For the tests I mainly imitated what I saw already implemented and made use of the variables ``test_eegieeg_data`` and ``test_converteeg_data``. 

I had to add some warning filters for cnt files inspired on:

```
cnt_ignore_warns = [
    pytest.mark.filterwarnings(
        'ignore:.*Could not parse meas date from the header. Setting to None.'
    ),
    pytest.mark.filterwarnings((
        'ignore:.*Could not define the number of bytes automatically.'
        ' Defaulting to 2.')
    ),
]
```

found in [this mne code](https://github.com/mne-tools/mne-python/blob/fd4f692f142ed8107cc816c3f60cfb3b4509e477/mne/channels/tests/test_montage.py#L1106-L1114)

I also had the problem of handedness set to None for the cnt file example. So I had to add also this warning filter:
``no_hand='ignore:.*Not setting subject handedness.:RuntimeWarning:mne'``

Maybe there is a better way to deal with that.

## make pep output:
```
Running flake8
0
Running pydocstyle
Checking MANIFEST.in
lists of files in version control and sdist match
```
## make test output:

```
======================= 250 passed in 317.49s (0:05:17) =======================
```

ci for docs-buikd is failing but maybe this is due to #941 , in correspondence my local build of the docs is also failing with the same error of the ci:

```
Extension error:
Here is a summary of the problems encountered when running the examples

Unexpected failing examples:
Y:\code\mne-bids\examples\read_bids_datasets.py failed leaving traceback:
Traceback (most recent call last):
  File "Y:\code\mne-bids\examples\read_bids_datasets.py", line 67, in <module>
    openneuro.download(dataset=dataset, target_dir=bids_root,
  File "C:\software\anaconda3\envs\mnedev\lib\site-packages\openneuro\download.py", line 602, in download
    local_tag = _get_local_tag(dataset_id=dataset,
  File "C:\software\anaconda3\envs\mnedev\lib\site-packages\openneuro\download.py", line 495, in _get_local_tag
    raise RuntimeError(f'The existing dataset in the target directory '
RuntimeError: The existing dataset in the target directory appears to be different from the one you requested to download. "DatasetDOI" field in local "dataset_description.json": doi:10.18112/openneuro.ds002778.v1.0.5. Requested dataset: ds002778
```

hoechenberger commented:
I just merged `main`, hopefully CI will now go green!

yjmantilla commented:
Im failing code coverage for some reason because of mne_bids/inspect.py although I didnt touch it. Otherwise it seems good

<html>
<body>
<!--StartFragment-->




mne_bids/config.py | 97.46% <ø> (ø) |  
mne_bids/inspect.py | 97.02% <66.66%> (-1.74%) | ⬇️
mne_bids/read.py | 95.84% <100.00%> (+0.02%) | ⬆️
mne_bids/write.py | 96.22% <100.00%> (ø)

<br class="Apple-interchange-newline"><!--EndFragment-->
</body>
</html>

sappelhoff reviewed (approved):
thanks @yjmantilla

agramfort reviewed (approved):
@hoechenberger merge if happy

hoechenberger commented:
Thanks, @yjmantilla!
